### ISSUE TITLE
ER: Developers are creating branches off hackforla/website

### ISSUE THREAD

**Issue by roslynwythe** [hackforla/website#4876]
### Dependency
- [ ] #6416 

### Emergent Requirement - Problem
Developers inadvertently create extraneous branches off the hackforla/website repository (instead of creating branches on their fork) but they do not have rights to delete the extraneous branches.  We need to prevent developers from creating these branches, and we require an issue template for instances for use when such a branch has been created and needs to be deleted.  

### Issue you discovered this emergent requirement in
- N/A

### Date discovered 
discussed at meeting 6/12/2023

### Did you have to do something temporarily
- [x] YES
- [ ] NO

We are currently pruning the branches manually, which is a waste of time.

### Who was involved
Bonnie Wolfe, @JessicaLucindaCheng 

### What happens if this is not addressed
The branches will persist until deleted by a lead and the lead will need to figure out which branches can be deleted, which can take a lot of time and research.

### Recommended Action Items
- [x] Make a new issue
- [ ] Discuss with team
- [ ] Let a Team Lead know

### Potential solutions [draft]
- [x] prevent write members who are not part of merge team from creating branches
   - @JessicaLucindaCheng implemented the solution for preventing members (aside from merge team) from creating branches off hackforla/website in the future by doing the following:
      <details>
        <summary>Click here to see the detailed instructions</summary>
        <ol>
          <li>On the repo, select <strong>Settings</strong>.</li>
          <li>On the left menu, select <strong>Branches</strong>.</li>
          <li>Select <strong>Edit</strong> for the branch rule named <strong>*</strong>. (Yes, the branch rule is named with the asterisk symbol).</li>
          <li>
             Scroll toward the bottom of the page. Then, you'll see the <em>Restrict pushes that create matching branches</em>; make sure the checkbox next to it is selected. <br>
            <img width="419" alt="restrict making branch" src="https://github.com/hackforla/website/assets/31293603/a8dcfd44-8de7-4fcd-8249-2bbfb8948694">
           </li>
        </ol>
      
      </details>

- [ ] Write an issue for a back-end developer to work on creating an issue template that includes both:
  - [ ] A section for reporting that a branch created on the Website repo needs to be removed
  - [ ] A section with instructions for how to remove the branch.
    - [ ] Instructions can be copied and pasted from [Insert link to instructions] into the issue template

**Comment by roslynwythe** [comment]
@JessicaLucindaCheng - I believe you have prevented future branches off hackforla/website.  Can you advise whether we still need this ER?

**Comment by JessicaLucindaCheng** [comment]
@roslynwythe 

> @JessicaLucindaCheng - I believe you have prevented future branches off hackforla/website. Can you advise whether we still need this ER?

Before I can answer that, I have a question. Since merge team members can create branches but I don't think they can delete them, how will we know when a branch created by a merge team member needs to be deleted?

**Comment by ExperimentsInHonesty** [comment]
@JessicaLucindaCheng We can create an issue template  I have updated the potential solution draft above and added check boxes.  Are you ready for me to prioritize this?

**Comment by JessicaLucindaCheng** [comment]
@ExperimentsInHonesty 

> @JessicaLucindaCheng We can create an issue template I have updated the potential solution draft above and added check boxes. Are you ready for me to prioritize this?

Yes, it's ready to be prioritized. Please note that I made some edits to the _Potential Solutions (draft)_ section and changed the feature label.

**Comment by ExperimentsInHonesty** [comment]
@JessicaLucindaCheng I updated the possible solutions template part.  From

>Write an issue for a back-end developer to work on creating an issue template for reporting that a branch created by the Merge Team needs to be removed. 

to 

>Write an issue for a back-end developer to work on creating an issue template for reporting that a branch created by the Merge Team needs to be removed. The issue template can be the report and the instructions for removing all in one.

If you agree, add ready for prioritization label, otherwise leave comment.

**Comment by JessicaLucindaCheng** [comment]
> @JessicaLucindaCheng I updated the possible solutions template part. From
> 
> > Write an issue for a back-end developer to work on creating an issue template for reporting that a branch created by the Merge Team needs to be removed.
> 
> to
> 
> > Write an issue for a back-end developer to work on creating an issue template for reporting that a branch created by the Merge Team needs to be removed. The issue template can be the report and the instructions for removing all in one.
> 
> If you agree, add ready for prioritization label, otherwise leave comment.

- I think the report and instructions can be included in the issue template together. However, I think a person on the website-admins team would have to write the instructions because I don't think other members of the team would necessarily know how within the context of the repo's branch rules set up. So, I wrote #6416.
- Also, I rewrote the instructions to be:
> - [ ] Write an issue for a back-end developer to work on creating an issue template that includes both:
>   - [ ] A section for reporting that a branch created on the Website repo needs to be removed
>   - [ ] A section with instructions for how to remove the branch.
>     - [ ] Instructions can be copied and pasted from [Insert link to instructions] into the issue template

**Comment by JessicaLucindaCheng** [comment]
@ExperimentsInHonesty 

> @JessicaLucindaCheng I updated the possible solutions template part. From
> 
> > Write an issue for a back-end developer to work on creating an issue template for reporting that a branch created by the Merge Team needs to be removed.
> 
> to
> 
> > Write an issue for a back-end developer to work on creating an issue template for reporting that a branch created by the Merge Team needs to be removed. The issue template can be the report and the instructions for removing all in one.
> 
> If you agree, add ready for prioritization label, otherwise leave comment.

- I think the report and instructions can be included in the issue template together. However, I think a person on the website-admins team would have to write the instructions because I don't think other members of the team would necessarily know how within the context of the repo's branch rules set up. So, I wrote #6416.
- Also, I rewrote the instructions to be:
> - [ ] Write an issue for a back-end developer to work on creating an issue template that includes both:
>   - [ ] A section for reporting that a branch created on the Website repo needs to be removed
>   - [ ] A section with instructions for how to remove the branch.
>     - [ ] Instructions can be copied and pasted from [Insert link to instructions] into the issue template

### Next steps
- [x] **Product:** 
  - [x] Prioritize #6416 
  - [x] For this ER (#4876), 
    - [x] Remove the `ready for product` label 
    - [x] Add a `ready for dev lead` label
    - [x] Move this ER into the Ice Box

**Comment by JessicaLucindaCheng** [comment]
### Merge Team member or Technical Lead
- [ ] Once the issue in the Dependency section of this ER is closed, review this ER and make it `ready for prioritization`. Add a `ready for prioritization` label.
